pkg/daemon/ceph/client/crush.go

```diff
@@ -165,7 +165,8 @@ func CrushReweight(context *clusterd.Context, clusterName string, id int, weight
 	args := []string{"osd", "crush", "reweight", fmt.Sprintf("osd.%d", id), fmt.Sprintf("%.1f", weight)}
 	buf, err := ExecuteCephCommand(context, clusterName, args)
 	if err != nil {
-		return "", fmt.Errorf("failed to crush reweight: %+v, %s", err, string(buf))
+		logger.Errorf("failed to crush reweight: %+v, %s", err, string(buf))
+		return "", err
```

Comment:
Not sure if this change to error handling is controversial (returning the original error and relying on its type elsewhere).  I couldn't see how to do this with the original code's practice of returning a string.

Reply:
@jcsp we should do more error checking for specific error codes where possible, agreed this is a good pattern. The most common place we do this is with the calls to the K8s api and we get back certain errors that we can check with the `errors` package, such as `errors.IsNotFound(err)`. We do commonly wrap the errors with a more descriptive context message, so when we need the original error code we would need to pay close attention that the original error is not getting wrapped.

Reply:
i agree that returning the original error type is needed here, because the caller will need to check values on it, but we are losing the `buf` value being returned now.  i think it would be better to return `string(buf), err` instead, so a caller has the option of doing something with string that is in `buf`.  If we are returning the information in `buf` once again, then I'd argue that logging within this function maybe should not be done and that should instead be the responsibility of the caller (if they want it logged).

Reply:
@jbw976 agreed, updated.